doc/plutus/explanations/order-book-pattern.rst

```diff
@@ -0,0 +1,119 @@
+
+This is exactly the information that we need for our dealings with the stablecoin, but how do we get it from the DEX to the stablecoin?
+There are two options.
+To choose the right one we need to consider the requirement and usage patterns of our application.
+
+1. Oracle UTXOs
+...............
+
+We could change the DEX contract to produce a new script UTXO for each fulfilled order that records the time and exchange rate of the order.
+The stablecoin user creates a UTXO with a script that requires an oracle UTXO to be present in the spending transaction, and the match maker would put the oracle UTXO and the stablecoin-action UTXO into one transaction and submit it.
+
+2. Oracle tokens
+................
+
+In a variation of the first idea, the DEX could produce *tokens* that encode the script-certified information we are interested in.
+We could set the asset name of the token to be the hash of the exchange rate data, and allow the transaction to produce any quantitiy of these tokens when the order is fulfilled.
+
+The :term:`minting policy <minting policy script>` of the oracle token should allow any number of tokens of the same asset name to be created as long as at least one token with that asset name exists already, reflecting the idea that information is hard to obtain but easy to replicate.
+
+The consumer of the oracle token needs to check that a token with the expected MPS hash is present in the transaction, and that the datum value of the token's asset name is available.
+Then it can use the information from the datum.
+Maybe it could even destroy the token when it has been used.
+
+This approach has the advantage of not clogging up the UTXO set too much, but the big question here is:
+How do we make the oracle token available to the match maker?
+It has to be stored in an output that the match maker can spend.
+Maybe the tokenomics of the exchange could have incentives to make this information flow to where it is needed.
+
+State machines
+~~~~~~~~~~~~~~
+
+State machines are a way of modeling smart contracts that is easy to understand and reason about.
+However, in their basic formulation they keep the entire state of each individual execution in a single UTXO, which puts them at risk of UTXO congestion caused by multiple users trying to transition the instance to a new state at the same time.
+
+If we can batch multiple transitions into one (for example, by finding a suitable `Semigroup` instance for the state machine's input type) then we could use the order book pattern to allow a number of users to submit transitions *without spending the UTXO* with the state machine instance's state.
+The match maker would construct a transaction that applies the sum of all proposed transitions in a single step.
+IOG is actively pursuing research in this area.
+
+Conclusion
+----------
+
+In the order book pattern we materialise *actions* as transaction outputs on the ledger, separating them from the state that they act on.
```

Comment:
I think this conclusion is really nice; and in fact _so_ nice that maybe it should be in the introduction! It seems to be a very nice way of explaining the general technique, and helps frame how to think about it.

Reply:
Thanks. I copied some of it to the introduction.